app/models/submission.rb

```diff
@@ -92,6 +92,12 @@ def reviews
       revision.reviews
     end
   end
+  
+  def authors_institutions
+    authors = self.authorships.all
+    authors_institutions = authors.flat_map{|e| e.person.find_current_institutions }.uniq
```

Comment:
Można pominąć przypisanie do tej zmiennej i zostawić to jako ostatnią linijkę tej metody.